crates/oxc_minifier/src/printer/gen.rs

```diff
@@ -502,7 +502,13 @@ impl<'a> Gen for FunctionBody<'a> {
         for directive in &self.directives {
             directive.gen(p);
         }
-        p.needs_semicolon = false;
+        p.needs_semicolon =
+            if let Some(Statement::ExpressionStatement(expr_stmt)) = self.statements.get(0)
+            && let Expression::StringLiteral(_) = &expr_stmt.expression {
```

Comment:
`self.statements.get(0).filter(|expr| matches!(expr, ...)`

Reply:
Oh, this is not going to work with box_patterns

then try `&& matches!(expr_stmt.expression, ...)`